mitodl/mitxpro
# Implement discount codes for B2B purchases

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #1047 

#### What's this PR do?
Implements discount codes for the bulk ecommerce page.

#### How should this be manually tested?
Go to the Django admin and create a B2BCoupon for a product. Use the coupon code you specified in the form and click 'Apply'. Once you have a quantity and product selected, the discount should show up. If the discount percent is 100% it should direct you to the receipt page automatically where you can download your enrollment codes.

#### Screenshots (if appropriate)
![Screenshot from 2019-09-04 16-23-49](https://user-images.githubusercontent.com/863262/64288588-7af88d00-cf30-11e9-8c53-f864b29cb991.png)

![Screenshot from 2019-09-04 16-24-17](https://user-images.githubusercontent.com/863262/64288592-7d5ae700-cf30-11e9-827c-df4740c527fa.png)
